src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -443,23 +443,24 @@ public static TypeCode GetTypeCode(Type? type)
         {
             if (RuntimeHelpers.IsKnownConstant(type))
             {
-                if (type is RuntimeType rt && rt.IsActualEnum)
-                    type = type.GetEnumUnderlyingType();
-                if (type == typeof(sbyte))
+                Type? underlyingType = type;
+                if (type is RuntimeType && ((RuntimeType)type).IsActualEnum)
```

Comment:
```suggestion
                if (((RuntimeType)type).IsActualEnum)
```
If `IsKnownConstant` returned true for types that are not `RuntimeType`s, this code would not be correct.

Reply:
It can also return true for `null` I think.